### PR TITLE
Fix Kafka CI

### PR DESCRIPTION
**Pull request by TimPansino**
# Overview

* Add `librdkafka` package from source to container for `confluent-kafka`.
* Reenable kafka testing on CI.
* Remove special case packages from kafka CI process.
* Adjust kafka db settings to force IPv4
* Small cleanup to pip scripting to match devcontainer.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -160,7 +160,7 @@ envlist =
     python-framework_tornado-{py38,py39,py310,py311}-tornadomaster,
     rabbitmq-messagebroker_pika-{py27,py37,py38,py39,pypy27,pypy38}-pika0.13,
     rabbitmq-messagebroker_pika-{py37,py38,py39,py310,py311,pypy38}-pikalatest,
-    kafka-messagebroker_confluentkafka-{py27,py37,py38,py39,py310,py311}-confluentkafkalatest,
+    kafka-messagebroker_confluentkafka-{py37,py38,py39,py310,py311}-confluentkafkalatest,
```
Comment (hmstepanek) [inline tox.ini:163]: Why are we dropping 2.7 testing here? It looks like it's still supported on the latest version.

**Review by hmstepanek — CHANGES_REQUESTED** [review]
Looks good-had one small question about 2.7 tests and a reminder about changing the tag to latest.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -119,7 +119,7 @@ jobs:
 
     runs-on: ubuntu-20.04
     container:
-      image: ghcr.io/newrelic/newrelic-python-agent-ci:sha-b0ffe8bdbb28ba0579377076ad680054da8fbc28
+      image: ghcr.io/newrelic/newrelic-python-agent-ci:sha-7f686305c0ddf52e731eb8ee91668ff73aeb7a03
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply .github/workflows/tests.yml:122]: Correct

File: tox.ini (hunk 1 above)
Comment (TimPansino) [reply tox.ini:163]: From my testing the library is broken on Python 2 now, it throws import errors on the latest versions.

Comment (hmstepanek) [reply tox.ini:163]: Oh ouch ok. Sounds good then!

**Review by hmstepanek — APPROVED** [review]
Assuming the tests all pass I'm pre-approving this. Just make sure we change the docker image sha over to latest before merging.

**Pull request merged**